### ISSUE TITLE
Load transport and persistence at runtime

### ISSUE THREAD

**Issue by mikeminutillo** [Particular/ServiceControl#3145]
The way things currently work, a transport and persistence are selected at install time and are copied in to the installation folder and a setting is created in the configuration file to determine which type to dynamically load at runtime.

This makes it harder to switch between (or even just select) configurations at run time because we need to copy in different assemblies and update the configuration file.

It would be handy to be able to switch between configurations for docker container images where we'd like to be able to select a transport using only environment variables.

This could be accomplished by copying the entire transport and peristence folder structure out of the packaging zip and into the installation folder, and then selecting an item at runtime.

i.e.

```
/ServiceControl.Audit.exe
/ServiceControl.Audit.exe.config
...
/Persistence/RavenDB35/persistence.manifest
/Persistence/RavenDB35/ServiceControl.Audit.Persistence.RavenDB35.dll
...
/Persistence/RavenDB5/persistence.manifest
/Persistence/RavenDB5/ServiceControl.Audit.Persistence.RavenDB5.dll
...
/Transports/ASBS/transport.manifest
/Transports/ASBS/ServiceControl.Transports.AzureServiceBus.dll
...
/Transports/MSMQ/transport.manifest
/Transports/MSMQ/ServiceControl.Transports.MSMQ.dll
...
```

Each manifest has a key indicating what it's called and the config file uses those keys to decide what gets loaded at runtime. i.e.

```xml
<configuration>
  <appSettings>
    <add key="ServiceControl.Audit/Persistence" value="RavenDB5" />
    <add key="ServiceControl.Audit/Transport" value="ASBS" />
  </appSettings>
</configuration>
```

And the docker configuration would be simplified:

[.env file](https://github.com/Particular/docs.particular.net/blob/master/samples/platformtools-docker-compose/ServiceControl_4/.env)
```
SERVICECONTROL_AUDIT_PERSISTENCE=RavenDB5
SERVICECONTROL_AUDIT_TRANSPORT=ASBS
```

This would also simplify the installation process as it no longer needs to selectively copy anything.

It could eventually lead to a separate lifecycle for transport and persistence options, the ability to configure which one gets used at runtime, the ability to use mutliple in a single instance (i.e. managing the audit queue of multiple transports).

**Comment by DavidBoike** [comment]
Related issue: complete the "plumbing" of the persistence.manifest file so that the information is available at runtime and returned in `/api/configuration` calls among multiple instances: https://github.com/Particular/ServiceControl/issues/3395

**Comment by mauroservienti** [comment]
This has been partially addressed in #3528. All instance types now support loading transports and persisters at runtime however, SCMU, PowerShell, and the installation engine have not been updated yet to support it.

**Comment by mauroservienti** [comment]
The next 5.2 release of ServiceControl supports all that mentioned in this issue.
